Index: plplot-5.9.7/cmake/modules/FindSWIG.cmake
===================================================================
--- plplot-5.9.7.orig/cmake/modules/FindSWIG.cmake	2011-01-15 16:00:41.000000000 +0100
+++ plplot-5.9.7/cmake/modules/FindSWIG.cmake	2011-01-15 16:01:09.000000000 +0100
@@ -26,7 +26,7 @@
     STRING(REGEX REPLACE "[\n\r]+" ";" SWIG_swiglib_output ${SWIG_swiglib_output})
     # force the path to be computed each time in case SWIG_EXECUTABLE has changed.
     SET(SWIG_DIR SWIG_DIR-NOTFOUND)
-    FIND_PATH(SWIG_DIR swig.swg PATHS ${SWIG_swiglib_output})
+    FIND_PATH(SWIG_DIR swig.swg PATHS ${SWIG_swiglib_output} NO_CMAKE_FIND_ROOT_PATH)
     IF(SWIG_DIR)
       SET(SWIG_FOUND 1)
       SET(SWIG_USE_FILE ${CMAKE_ROOT}/Modules/UseSWIG.cmake)
