Index: plplot-5.9.7/bindings/octave/CMakeLists.txt
===================================================================
--- plplot-5.9.7.orig/bindings/octave/CMakeLists.txt	2011-01-04 17:02:32.000000000 +0100
+++ plplot-5.9.7/bindings/octave/CMakeLists.txt	2011-01-04 17:03:09.000000000 +0100
@@ -222,7 +222,7 @@
     COMMAND cat ${CMAKE_CURRENT_SOURCE_DIR}/plplot_octave_def >> plplot_stub.m
     COMMAND echo >> plplot_stub.m
     COMMAND 
-    ${massage_LOCATION} >> plplot_stub.m 2> missing_help
+    plplot-massage >> plplot_stub.m 2> missing_help
     DEPENDS
     ${CMAKE_CURRENT_BINARY_DIR}/tmp_stub
     ${CMAKE_CURRENT_SOURCE_DIR}/plplot_octave_def
