Index: plplot-5.9.7/cmake/modules/plplot.cmake
===================================================================
--- plplot-5.9.7.orig/cmake/modules/plplot.cmake	2011-05-26 20:25:37.000000000 +0200
+++ plplot-5.9.7/cmake/modules/plplot.cmake	2011-05-26 20:25:37.000000000 +0200
@@ -493,7 +493,8 @@
 # =======================================================================
 # libpango support
 # =======================================================================
-include(pango)
+# pango disabled for now.  compilation fails on openwrt (?).
+#include(pango)
 
 # =======================================================================
 # Device drivers
