diff -ur qstardict-0.13.1.or/qstardict/keyboard.cpp qstardict-0.13.1/qstardict/keyboard.cpp
--- qstardict-0.13.1.or/qstardict/keyboard.cpp	2009-02-10 15:33:30.000000000 +0300
+++ qstardict-0.13.1/qstardict/keyboard.cpp	2010-12-07 21:52:31.613898108 +0300
@@ -82,7 +82,28 @@
 
 } // namespace
 
-#endif // Q_WS_WIN
+#elif defined(Q_WS_QWS) // Q_WS_WIN
+
+namespace
+{
+const unsigned mAlt     = 0010;
+const unsigned mCtrl    = 0004;
+const unsigned mShift   = 0001;
+const unsigned mWin     = 0100;
+}
+
+namespace QStarDict
+{
+
+Qt::KeyboardModifiers Keyboard::activeModifiers()
+{
+    Qt::KeyboardModifiers result;
+    return result;
+}
+
+} // namespace
+
+#endif // Q_WS_QWS
 
 // vim: tabstop=4 softtabstop=4 shiftwidth=4 expandtab cindent textwidth=120 formatoptions=tc
 
diff -ur qstardict-0.13.1.or/qstardict/mainwindow.ui qstardict-0.13.1/qstardict/mainwindow.ui
--- qstardict-0.13.1.or/qstardict/mainwindow.ui	2009-02-10 15:33:30.000000000 +0300
+++ qstardict-0.13.1/qstardict/mainwindow.ui	2010-12-08 16:40:21.843773360 +0300
@@ -5,8 +5,8 @@
    <rect>
     <x>0</x>
     <y>0</y>
-    <width>742</width>
-    <height>511</height>
+    <width>320</width>
+    <height>240</height>
    </rect>
   </property>
   <property name="windowTitle" >
@@ -19,15 +19,15 @@
   <widget class="QWidget" name="centralwidget" >
    <layout class="QGridLayout" >
     <property name="margin" >
-     <number>9</number>
+     <number>0</number>
     </property>
     <property name="spacing" >
-     <number>6</number>
+     <number>0</number>
     </property>
     <item row="0" column="0" >
      <layout class="QHBoxLayout" >
       <property name="spacing" >
-       <number>6</number>
+       <number>0</number>
       </property>
       <property name="margin" >
        <number>0</number>
@@ -90,17 +90,16 @@
     <number>1</number>
    </attribute>
    <widget class="QWidget" name="dockWidgetContents" >
-    <layout class="QGridLayout" >
-     <property name="margin" >
-      <number>9</number>
-     </property>
-     <property name="spacing" >
-      <number>6</number>
+    <widget class="QListWidget" name="wordsList" >
+     <property name="geometry">
+      <rect>
+       <x>9</x>
+       <y>9</y>
+       <width>185</width>
+       <height>184</height>
+      </rect>
      </property>
-     <item row="0" column="0" >
-      <widget class="QListWidget" name="wordsList" />
-     </item>
-    </layout>
+    </widget>
    </widget>
   </widget>
   <widget class="QMenuBar" name="menubar" >
@@ -108,8 +107,8 @@
     <rect>
      <x>0</x>
      <y>0</y>
-     <width>742</width>
-     <height>22</height>
+     <width>320</width>
+     <height>18</height>
     </rect>
    </property>
    <widget class="QMenu" name="menu_File" >
diff -ur qstardict-0.13.1.or/qstardict/settingsdialog.ui qstardict-0.13.1/qstardict/settingsdialog.ui
--- qstardict-0.13.1.or/qstardict/settingsdialog.ui	2009-02-10 15:33:30.000000000 +0300
+++ qstardict-0.13.1/qstardict/settingsdialog.ui	2010-12-08 10:27:03.160773247 +0300
@@ -5,8 +5,8 @@
    <rect>
     <x>0</x>
     <y>0</y>
-    <width>463</width>
-    <height>498</height>
+    <width>320</width>
+    <height>240</height>
    </rect>
   </property>
   <property name="windowTitle" >
@@ -45,8 +45,8 @@
        <rect>
         <x>0</x>
         <y>0</y>
-        <width>441</width>
-        <height>414</height>
+        <width>320</width>
+        <height>240</height>
        </rect>
       </property>
       <attribute name="title" >
@@ -104,8 +104,8 @@
        <rect>
         <x>0</x>
         <y>0</y>
-        <width>441</width>
-        <height>414</height>
+        <width>320</width>
+        <height>240</height>
        </rect>
       </property>
       <attribute name="title" >
@@ -282,8 +282,8 @@
        <rect>
         <x>0</x>
         <y>0</y>
-        <width>640</width>
-        <height>409</height>
+        <width>320</width>
+        <height>240</height>
        </rect>
       </property>
       <attribute name="title" >
@@ -565,8 +565,8 @@
        <rect>
         <x>0</x>
         <y>0</y>
-        <width>640</width>
-        <height>409</height>
+        <width>320</width>
+        <height>240</height>
        </rect>
       </property>
       <attribute name="title" >
