diff --git a/plugins/stardict/dictziplib.cpp b/plugins/stardict/dictziplib.cpp
index 867e330..bb62a54 100644
--- a/plugins/stardict/dictziplib.cpp
+++ b/plugins/stardict/dictziplib.cpp
@@ -305,7 +305,7 @@ bool dictData::open(const std::string& fname, int computeCRC)
 
 #ifdef Q_OS_UNIX
     if (stat(fname.c_str(), &sb) || !S_ISREG(sb.st_mode))
-#elif def Q_OS_WIN32
+#elif defined Q_OS_WIN32
     if (_stat(fname.c_str(), &sb) || !(sb.stMode & _S_IFREG))
 #endif
     {
diff --git a/qstardict/keyboard.cpp b/qstardict/keyboard.cpp
index cc6e406..5a6d514 100644
--- a/qstardict/keyboard.cpp
+++ b/qstardict/keyboard.cpp
@@ -82,7 +82,28 @@ Qt::KeyboardModifiers Keyboard::activeModifiers()
 
 } // namespace
 
-#endif // Q_WS_WIN
+#elif defined(Q_WS_QWS) // Q_WS_WIN
+
+namespace
+{
+const unsigned mAlt     = 0010;
+const unsigned mCtrl    = 0004;
+const unsigned mShift   = 0001;
+const unsigned mWin     = 0100;
+}
+
+namespace QStarDict
+{
+
+Qt::KeyboardModifiers Keyboard::activeModifiers()
+{
+    Qt::KeyboardModifiers result;
+    return result;
+}
+
+} // namespace
+
+#endif // Q_WS_QWS
 
 // vim: tabstop=4 softtabstop=4 shiftwidth=4 expandtab cindent textwidth=120 formatoptions=tc
 
diff --git a/qstardict/main.cpp b/qstardict/main.cpp
index 8a13dc9..a726364 100644
--- a/qstardict/main.cpp
+++ b/qstardict/main.cpp
@@ -35,6 +35,10 @@
 #include <QTranslator>
 #endif // QSTARDICT_WITH_TRANSLATIONS
 
+#ifdef Q_WS_QWS
+#include <QtGui/QWSServer>
+#endif
+
 int main(int argc, char *argv[])
 {
     QStarDict::Application app(argc, argv);
@@ -61,6 +65,9 @@ int main(int argc, char *argv[])
     }
 #endif // Q_OS_WIN
 
+#ifdef Q_WS_QWS
+        QWSServer::setCursorVisible(false);
+#endif
 
     return app.exec();
 }
diff --git a/qstardict/mainwindow.cpp b/qstardict/mainwindow.cpp
index e8b9456..29ac486 100644
--- a/qstardict/mainwindow.cpp
+++ b/qstardict/mainwindow.cpp
@@ -93,7 +93,7 @@ void MainWindow::loadSettings()
     setVisible(config.value("MainWindow/visible", true).toBool());
     wordsListDock->setFloating(config.value("MainWindow/wordsListDock/floating", wordsListDock->isFloating()).toBool());
     wordsListDock->setGeometry(config.value("MainWindow/wordsListDock/geometry", wordsListDock->geometry()).toRect());
-    setInstantSearch(config.value("MainWindow/instantSearch", true).toBool());
+    setInstantSearch(config.value("MainWindow/instantSearch", false).toBool());
     setDefaultStyleSheet(config.value("MainWindow/defaultStyleSheet", defaultStyleSheet()).toString());
 }
 
diff --git a/qstardict/mainwindow.ui b/qstardict/mainwindow.ui
index 68a37bf..fcabb84 100644
--- a/qstardict/mainwindow.ui
+++ b/qstardict/mainwindow.ui
@@ -1,190 +1,241 @@
-<ui version="4.0" >
+<?xml version="1.0" encoding="UTF-8"?>
+<ui version="4.0">
  <class>QStarDict::MainWindow</class>
- <widget class="QMainWindow" name="QStarDict::MainWindow" >
-  <property name="geometry" >
+ <widget class="QMainWindow" name="QStarDict::MainWindow">
+  <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
-    <width>742</width>
-    <height>511</height>
+    <width>326</width>
+    <height>240</height>
    </rect>
   </property>
-  <property name="windowTitle" >
+  <property name="cursor">
+   <cursorShape>BlankCursor</cursorShape>
+  </property>
+  <property name="windowTitle">
    <string>QStarDict</string>
   </property>
-  <property name="windowIcon" >
-   <iconset resource="pixmaps/pixmaps.qrc" >
+  <property name="windowIcon">
+   <iconset resource="pixmaps/pixmaps.qrc">
     <normaloff>:/icons/qstardict.png</normaloff>:/icons/qstardict.png</iconset>
   </property>
-  <widget class="QWidget" name="centralwidget" >
-   <layout class="QGridLayout" >
-    <property name="margin" >
-     <number>9</number>
+  <widget class="QWidget" name="centralwidget">
+   <property name="minimumSize">
+    <size>
+     <width>214</width>
+     <height>130</height>
+    </size>
+   </property>
+   <property name="maximumSize">
+    <size>
+     <width>320</width>
+     <height>240</height>
+    </size>
+   </property>
+   <layout class="QVBoxLayout" name="verticalLayout_2">
+    <property name="spacing">
+     <number>0</number>
     </property>
-    <property name="spacing" >
-     <number>6</number>
+    <property name="margin">
+     <number>0</number>
     </property>
-    <item row="0" column="0" >
-     <layout class="QHBoxLayout" >
-      <property name="spacing" >
-       <number>6</number>
+    <item>
+     <layout class="QHBoxLayout">
+      <property name="spacing">
+       <number>0</number>
       </property>
-      <property name="margin" >
+      <property name="margin">
        <number>0</number>
       </property>
       <item>
-       <widget class="QToolButton" name="searchClearButton" >
-        <property name="toolTip" >
+       <widget class="QToolButton" name="searchClearButton">
+        <property name="toolTip">
          <string>Clear the search box</string>
         </property>
-        <property name="text" >
+        <property name="text">
          <string>Clear</string>
         </property>
-        <property name="icon" >
-         <iconset resource="pixmaps/pixmaps.qrc" >
+        <property name="icon">
+         <iconset resource="pixmaps/pixmaps.qrc">
           <normaloff>:/icons/clear-right.png</normaloff>:/icons/clear-right.png</iconset>
         </property>
-        <property name="toolButtonStyle" >
+        <property name="toolButtonStyle">
          <enum>Qt::ToolButtonIconOnly</enum>
         </property>
        </widget>
       </item>
       <item>
-       <widget class="QLineEdit" name="searchBox" />
+       <widget class="QLineEdit" name="searchBox"/>
       </item>
       <item>
-       <widget class="QToolButton" name="queryButton" >
-        <property name="toolTip" >
+       <widget class="QToolButton" name="queryButton">
+        <property name="toolTip">
          <string>Fuzzy query</string>
         </property>
-        <property name="text" >
+        <property name="text">
          <string>Search</string>
         </property>
-        <property name="icon" >
-         <iconset resource="pixmaps/pixmaps.qrc" >
+        <property name="icon">
+         <iconset resource="pixmaps/pixmaps.qrc">
           <normaloff>:/icons/search-filter.png</normaloff>:/icons/search-filter.png</iconset>
         </property>
-        <property name="toolButtonStyle" >
+        <property name="toolButtonStyle">
          <enum>Qt::ToolButtonTextBesideIcon</enum>
         </property>
        </widget>
       </item>
      </layout>
     </item>
-    <item row="1" column="0" >
-     <widget class="DictWidget" native="1" name="translationView" />
+    <item>
+     <widget class="DictWidget" name="translationView" native="true"/>
     </item>
    </layout>
   </widget>
-  <widget class="QDockWidget" name="wordsListDock" >
-   <property name="features" >
-    <set>QDockWidget::AllDockWidgetFeatures</set>
+  <widget class="QDockWidget" name="wordsListDock">
+   <property name="minimumSize">
+    <size>
+     <width>100</width>
+     <height>155</height>
+    </size>
+   </property>
+   <property name="maximumSize">
+    <size>
+     <width>100</width>
+     <height>240</height>
+    </size>
    </property>
-   <property name="allowedAreas" >
+   <property name="features">
+    <set>QDockWidget::DockWidgetClosable|QDockWidget::DockWidgetMovable</set>
+   </property>
+   <property name="allowedAreas">
     <set>Qt::NoDockWidgetArea</set>
    </property>
-   <property name="windowTitle" >
+   <property name="windowTitle">
     <string>Words list</string>
    </property>
-   <attribute name="dockWidgetArea" >
+   <attribute name="dockWidgetArea">
     <number>1</number>
    </attribute>
-   <widget class="QWidget" name="dockWidgetContents" >
-    <layout class="QGridLayout" >
-     <property name="margin" >
-      <number>9</number>
+   <widget class="QWidget" name="dockWidgetContents">
+    <property name="minimumSize">
+     <size>
+      <width>100</width>
+      <height>130</height>
+     </size>
+    </property>
+    <property name="maximumSize">
+     <size>
+      <width>100</width>
+      <height>240</height>
+     </size>
+    </property>
+    <widget class="QListWidget" name="wordsList">
+     <property name="geometry">
+      <rect>
+       <x>0</x>
+       <y>5</y>
+       <width>100</width>
+       <height>190</height>
+      </rect>
+     </property>
+     <property name="minimumSize">
+      <size>
+       <width>100</width>
+       <height>130</height>
+      </size>
      </property>
-     <property name="spacing" >
-      <number>6</number>
+     <property name="maximumSize">
+      <size>
+       <width>100</width>
+       <height>240</height>
+      </size>
      </property>
-     <item row="0" column="0" >
-      <widget class="QListWidget" name="wordsList" />
-     </item>
-    </layout>
+    </widget>
    </widget>
   </widget>
-  <widget class="QMenuBar" name="menubar" >
-   <property name="geometry" >
+  <widget class="QMenuBar" name="menubar">
+   <property name="geometry">
     <rect>
      <x>0</x>
      <y>0</y>
-     <width>742</width>
-     <height>22</height>
+     <width>326</width>
+     <height>27</height>
     </rect>
    </property>
-   <widget class="QMenu" name="menu_File" >
-    <property name="title" >
+   <widget class="QMenu" name="menu_File">
+    <property name="title">
      <string>&amp;File</string>
     </property>
-    <addaction name="actionQuit" />
+    <addaction name="actionQuit"/>
    </widget>
-   <widget class="QMenu" name="menuHelp" >
-    <property name="title" >
+   <widget class="QMenu" name="menuHelp">
+    <property name="title">
      <string>&amp;Help</string>
     </property>
-    <addaction name="actionHelp" />
-    <addaction name="actionAbout" />
-    <addaction name="actionAboutQt" />
+    <addaction name="actionHelp"/>
+    <addaction name="actionAbout"/>
+    <addaction name="actionAboutQt"/>
    </widget>
-   <widget class="QMenu" name="menu_Options" >
-    <property name="title" >
+   <widget class="QMenu" name="menu_Options">
+    <property name="title">
      <string>&amp;Settings</string>
     </property>
-    <addaction name="actionScan" />
-    <addaction name="actionSettings" />
+    <addaction name="actionScan"/>
+    <addaction name="actionSettings"/>
    </widget>
-   <addaction name="menu_File" />
-   <addaction name="menu_Options" />
-   <addaction name="menuHelp" />
+   <addaction name="menu_File"/>
+   <addaction name="menu_Options"/>
+   <addaction name="menuHelp"/>
   </widget>
-  <action name="actionQuit" >
-   <property name="icon" >
-    <iconset resource="pixmaps/pixmaps.qrc" >
+  <action name="actionQuit">
+   <property name="icon">
+    <iconset resource="pixmaps/pixmaps.qrc">
      <normaloff>:/icons/application-exit.png</normaloff>:/icons/application-exit.png</iconset>
    </property>
-   <property name="text" >
+   <property name="text">
     <string>&amp;Quit</string>
    </property>
-   <property name="shortcut" >
+   <property name="shortcut">
     <string>Ctrl+Q</string>
    </property>
   </action>
-  <action name="actionAbout" >
-   <property name="text" >
+  <action name="actionAbout">
+   <property name="text">
     <string>&amp;About</string>
    </property>
   </action>
-  <action name="actionAboutQt" >
-   <property name="text" >
+  <action name="actionAboutQt">
+   <property name="text">
     <string>About &amp;Qt</string>
    </property>
   </action>
-  <action name="actionSettings" >
-   <property name="icon" >
-    <iconset resource="pixmaps/pixmaps.qrc" >
+  <action name="actionSettings">
+   <property name="icon">
+    <iconset resource="pixmaps/pixmaps.qrc">
      <normaloff>:/icons/configure.png</normaloff>:/icons/configure.png</iconset>
    </property>
-   <property name="text" >
+   <property name="text">
     <string>&amp;Configure QStarDict</string>
    </property>
   </action>
-  <action name="actionScan" >
-   <property name="checkable" >
+  <action name="actionScan">
+   <property name="checkable">
     <bool>true</bool>
    </property>
-   <property name="text" >
+   <property name="text">
     <string>&amp;Scan</string>
    </property>
   </action>
-  <action name="actionHelp" >
-   <property name="icon" >
-    <iconset resource="pixmaps/pixmaps.qrc" >
+  <action name="actionHelp">
+   <property name="icon">
+    <iconset resource="pixmaps/pixmaps.qrc">
      <normaloff>:/icons/help-contents.png</normaloff>:/icons/help-contents.png</iconset>
    </property>
-   <property name="text" >
+   <property name="text">
     <string>QStarDict &amp;Help</string>
    </property>
-   <property name="shortcut" >
+   <property name="shortcut">
     <string>F1</string>
    </property>
   </action>
@@ -203,7 +254,7 @@
   <tabstop>searchClearButton</tabstop>
  </tabstops>
  <resources>
-  <include location="pixmaps/pixmaps.qrc" />
+  <include location="pixmaps/pixmaps.qrc"/>
  </resources>
  <connections>
   <connection>
@@ -212,11 +263,11 @@
    <receiver>searchBox</receiver>
    <slot>clear()</slot>
    <hints>
-    <hint type="sourcelabel" >
+    <hint type="sourcelabel">
      <x>315</x>
      <y>59</y>
     </hint>
-    <hint type="destinationlabel" >
+    <hint type="destinationlabel">
      <x>647</x>
      <y>61</y>
     </hint>
@@ -228,11 +279,11 @@
    <receiver>searchBox</receiver>
    <slot>setFocus()</slot>
    <hints>
-    <hint type="sourcelabel" >
+    <hint type="sourcelabel">
      <x>315</x>
      <y>59</y>
     </hint>
-    <hint type="destinationlabel" >
+    <hint type="destinationlabel">
      <x>647</x>
      <y>61</y>
     </hint>
@@ -244,11 +295,11 @@
    <receiver>queryButton</receiver>
    <slot>click()</slot>
    <hints>
-    <hint type="sourcelabel" >
+    <hint type="sourcelabel">
      <x>438</x>
      <y>50</y>
     </hint>
-    <hint type="destinationlabel" >
+    <hint type="destinationlabel">
      <x>661</x>
      <y>51</y>
     </hint>
diff --git a/qstardict/settingsdialog.ui b/qstardict/settingsdialog.ui
index 357f15b..d08fbd5 100644
--- a/qstardict/settingsdialog.ui
+++ b/qstardict/settingsdialog.ui
@@ -1,84 +1,67 @@
-<ui version="4.0" >
+<?xml version="1.0" encoding="UTF-8"?>
+<ui version="4.0">
  <class>QStarDict::SettingsDialog</class>
- <widget class="QDialog" name="QStarDict::SettingsDialog" >
-  <property name="geometry" >
+ <widget class="QDialog" name="QStarDict::SettingsDialog">
+  <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
-    <width>463</width>
-    <height>498</height>
+    <width>320</width>
+    <height>240</height>
    </rect>
   </property>
-  <property name="windowTitle" >
+  <property name="sizePolicy">
+   <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
+    <horstretch>0</horstretch>
+    <verstretch>0</verstretch>
+   </sizepolicy>
+  </property>
+  <property name="cursor">
+   <cursorShape>BlankCursor</cursorShape>
+  </property>
+  <property name="windowTitle">
    <string>QStarDict Settings</string>
   </property>
-  <property name="toolTip" >
+  <property name="toolTip">
    <string/>
   </property>
-  <layout class="QGridLayout" >
-   <property name="margin" >
-    <number>9</number>
-   </property>
-   <property name="spacing" >
-    <number>6</number>
-   </property>
-   <item row="1" column="0" >
-    <widget class="QDialogButtonBox" name="buttonBox" >
-     <property name="orientation" >
-      <enum>Qt::Horizontal</enum>
-     </property>
-     <property name="standardButtons" >
-      <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
-     </property>
-     <property name="centerButtons" >
-      <bool>false</bool>
-     </property>
-    </widget>
-   </item>
-   <item row="0" column="0" >
-    <widget class="QTabWidget" name="tabWidget" >
-     <property name="currentIndex" >
+  <layout class="QGridLayout" name="gridLayout">
+   <item row="0" column="0">
+    <widget class="QTabWidget" name="tabWidget">
+     <property name="currentIndex">
       <number>0</number>
      </property>
-     <widget class="QWidget" name="tabGlobalSettings" >
-      <property name="geometry" >
-       <rect>
-        <x>0</x>
-        <y>0</y>
-        <width>441</width>
-        <height>414</height>
-       </rect>
-      </property>
-      <attribute name="title" >
+     <widget class="QWidget" name="tabGlobalSettings">
+      <attribute name="title">
        <string>Global settings</string>
       </attribute>
-      <layout class="QVBoxLayout" >
+      <layout class="QVBoxLayout">
        <item>
-        <widget class="QCheckBox" name="instantSearchBox" >
-         <property name="text" >
+        <widget class="QCheckBox" name="instantSearchBox">
+         <property name="text">
           <string>Instant search</string>
          </property>
         </widget>
        </item>
        <item>
-        <layout class="QHBoxLayout" >
-         <property name="spacing" >
+        <layout class="QVBoxLayout" name="verticalLayout_2">
+         <property name="spacing">
           <number>6</number>
          </property>
-         <property name="margin" >
+         <property name="margin">
           <number>0</number>
          </property>
          <item>
-          <widget class="QLabel" name="label_7" >
-           <property name="text" >
+          <widget class="QLabel" name="label_7">
+           <property name="text">
             <string>Pronounce words using this command:</string>
            </property>
           </widget>
          </item>
          <item>
-          <widget class="QLineEdit" name="speechCmdEdit" >
-           <property name="toolTip" >
-            <string>Enter cmd for the speaching program.&lt;br>If cmd contains "%s" it will be replaced to word, else word will be writen to stdin of speech process.</string>
+          <widget class="QLineEdit" name="speechCmdEdit">
+           <property name="toolTip">
+            <string>Enter cmd for the speaching program.&lt;br&gt;If cmd contains &quot;%s&quot; it will be replaced to word, else word will be writen to stdin of speech process.</string>
            </property>
           </widget>
          </item>
@@ -86,104 +69,96 @@
        </item>
        <item>
         <spacer>
-         <property name="orientation" >
+         <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
-         <property name="sizeHint" stdset="0" >
+         <property name="sizeHint" stdset="0">
           <size>
-           <width>369</width>
-           <height>171</height>
+           <width>320</width>
+           <height>240</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </widget>
-     <widget class="QWidget" name="tabDictionaries" >
-      <property name="geometry" >
-       <rect>
-        <x>0</x>
-        <y>0</y>
-        <width>441</width>
-        <height>414</height>
-       </rect>
-      </property>
-      <attribute name="title" >
+     <widget class="QWidget" name="tabDictionaries">
+      <attribute name="title">
        <string>Dictionaries</string>
       </attribute>
-      <layout class="QVBoxLayout" >
+      <layout class="QVBoxLayout">
        <item>
-        <widget class="QGroupBox" name="groupBox_3" >
-         <property name="title" >
+        <widget class="QGroupBox" name="groupBox_3">
+         <property name="title">
           <string>Dictionaries</string>
          </property>
-         <layout class="QVBoxLayout" >
+         <layout class="QVBoxLayout">
           <item>
-           <widget class="QTableView" name="dictsTableView" >
-            <property name="editTriggers" >
+           <widget class="QTableView" name="dictsTableView">
+            <property name="editTriggers">
              <set>QAbstractItemView::NoEditTriggers</set>
             </property>
-            <property name="selectionMode" >
+            <property name="selectionMode">
              <enum>QAbstractItemView::SingleSelection</enum>
             </property>
-            <property name="selectionBehavior" >
+            <property name="selectionBehavior">
              <enum>QAbstractItemView::SelectRows</enum>
             </property>
-            <property name="showGrid" >
+            <property name="showGrid">
              <bool>false</bool>
             </property>
            </widget>
           </item>
           <item>
-           <layout class="QHBoxLayout" >
+           <layout class="QHBoxLayout">
             <item>
-             <widget class="QToolButton" name="dictsMoveUpButton" >
-              <property name="toolTip" >
+             <widget class="QToolButton" name="dictsMoveUpButton">
+              <property name="toolTip">
                <string>Move up</string>
               </property>
-              <property name="text" >
+              <property name="text">
                <string>Up</string>
               </property>
-              <property name="icon" >
-               <iconset resource="pixmaps/pixmaps.qrc" >
+              <property name="icon">
+               <iconset resource="../plugins/stardict/pixmaps/pixmaps.qrc">
                 <normaloff>:/icons/arrow-up.png</normaloff>:/icons/arrow-up.png</iconset>
               </property>
              </widget>
             </item>
             <item>
-             <widget class="QToolButton" name="dictsMoveDownButton" >
-              <property name="toolTip" >
+             <widget class="QToolButton" name="dictsMoveDownButton">
+              <property name="toolTip">
                <string>Move down</string>
               </property>
-              <property name="text" >
+              <property name="text">
                <string>Down</string>
               </property>
-              <property name="icon" >
-               <iconset resource="pixmaps/pixmaps.qrc" >
+              <property name="icon">
+               <iconset resource="../plugins/stardict/pixmaps/pixmaps.qrc">
                 <normaloff>:/icons/arrow-down.png</normaloff>:/icons/arrow-down.png</iconset>
               </property>
              </widget>
             </item>
             <item>
-             <widget class="QToolButton" name="dictsShowInfoButton" >
-              <property name="toolTip" >
+             <widget class="QToolButton" name="dictsShowInfoButton">
+              <property name="toolTip">
                <string>Show information about dictionary</string>
               </property>
-              <property name="text" >
+              <property name="text">
                <string>Show info</string>
               </property>
-              <property name="icon" >
-               <iconset resource="pixmaps/pixmaps.qrc" >
+              <property name="icon">
+               <iconset resource="pixmaps/pixmaps.qrc">
                 <normaloff>:/icons/dialog-information.png</normaloff>:/icons/dialog-information.png</iconset>
               </property>
              </widget>
             </item>
             <item>
              <spacer>
-              <property name="orientation" >
+              <property name="orientation">
                <enum>Qt::Horizontal</enum>
               </property>
-              <property name="sizeHint" stdset="0" >
+              <property name="sizeHint" stdset="0">
                <size>
                 <width>40</width>
                 <height>20</height>
@@ -196,73 +171,80 @@
          </layout>
         </widget>
        </item>
+      </layout>
+     </widget>
+     <widget class="QWidget" name="tabPlugins">
+      <attribute name="title">
+       <string>Plugins</string>
+      </attribute>
+      <layout class="QVBoxLayout" name="verticalLayout">
        <item>
-        <widget class="QGroupBox" name="groupBox_4" >
-         <property name="maximumSize" >
+        <widget class="QGroupBox" name="groupBox_5">
+         <property name="maximumSize">
           <size>
            <width>16777215</width>
-           <height>196</height>
+           <height>16777215</height>
           </size>
          </property>
-         <property name="title" >
+         <property name="title">
           <string>Plugins</string>
          </property>
-         <property name="alignment" >
+         <property name="alignment">
           <set>Qt::AlignBottom|Qt::AlignLeading|Qt::AlignLeft</set>
          </property>
-         <layout class="QVBoxLayout" >
+         <layout class="QVBoxLayout" name="_2">
           <item>
-           <widget class="QTableView" name="pluginsTableView" >
-            <property name="editTriggers" >
+           <widget class="QTableView" name="pluginsTableView">
+            <property name="editTriggers">
              <set>QAbstractItemView::NoEditTriggers</set>
             </property>
-            <property name="selectionMode" >
+            <property name="selectionMode">
              <enum>QAbstractItemView::SingleSelection</enum>
             </property>
-            <property name="selectionBehavior" >
+            <property name="selectionBehavior">
              <enum>QAbstractItemView::SelectRows</enum>
             </property>
-            <property name="showGrid" >
+            <property name="showGrid">
              <bool>false</bool>
             </property>
            </widget>
           </item>
           <item>
-           <layout class="QHBoxLayout" >
+           <layout class="QHBoxLayout" name="_3">
             <item>
-             <widget class="QToolButton" name="pluginsShowInfoButton" >
-              <property name="toolTip" >
+             <widget class="QToolButton" name="pluginsShowInfoButton">
+              <property name="toolTip">
                <string>Show information about plugin</string>
               </property>
-              <property name="text" >
+              <property name="text">
                <string>Info</string>
               </property>
-              <property name="icon" >
-               <iconset resource="pixmaps/pixmaps.qrc" >
+              <property name="icon">
+               <iconset resource="pixmaps/pixmaps.qrc">
                 <normaloff>:/icons/dialog-information.png</normaloff>:/icons/dialog-information.png</iconset>
               </property>
              </widget>
             </item>
             <item>
-             <widget class="QToolButton" name="pluginsConfigureButton" >
-              <property name="toolTip" >
+             <widget class="QToolButton" name="pluginsConfigureButton">
+              <property name="toolTip">
                <string>Configure plugin</string>
               </property>
-              <property name="text" >
+              <property name="text">
                <string>Configure</string>
               </property>
-              <property name="icon" >
-               <iconset resource="pixmaps/pixmaps.qrc" >
+              <property name="icon">
+               <iconset resource="pixmaps/pixmaps.qrc">
                 <normaloff>:/icons/configure.png</normaloff>:/icons/configure.png</iconset>
               </property>
              </widget>
             </item>
             <item>
              <spacer>
-              <property name="orientation" >
+              <property name="orientation">
                <enum>Qt::Horizontal</enum>
               </property>
-              <property name="sizeHint" stdset="0" >
+              <property name="sizeHint" stdset="0">
                <size>
                 <width>40</width>
                 <height>20</height>
@@ -277,136 +259,128 @@
        </item>
       </layout>
      </widget>
-     <widget class="QWidget" name="tabPopupWindow" >
-      <property name="geometry" >
-       <rect>
-        <x>0</x>
-        <y>0</y>
-        <width>640</width>
-        <height>409</height>
-       </rect>
-      </property>
-      <attribute name="title" >
-       <string>Popup window</string>
+     <widget class="QWidget" name="tabPopupWindow">
+      <attribute name="title">
+       <string>Behaviour</string>
       </attribute>
-      <layout class="QGridLayout" >
-       <property name="margin" >
+      <layout class="QGridLayout">
+       <property name="margin">
         <number>9</number>
        </property>
-       <property name="spacing" >
+       <property name="spacing">
         <number>6</number>
        </property>
-       <item row="0" column="0" >
-        <widget class="QGroupBox" name="groupBox" >
-         <property name="title" >
+       <item row="0" column="0">
+        <widget class="QGroupBox" name="groupBox">
+         <property name="title">
           <string>Behavior</string>
          </property>
-         <layout class="QGridLayout" >
-          <property name="margin" >
+         <layout class="QGridLayout">
+          <property name="margin">
            <number>9</number>
           </property>
-          <property name="spacing" >
+          <property name="spacing">
            <number>6</number>
           </property>
-          <item row="4" column="0" >
-           <layout class="QVBoxLayout" >
-            <property name="spacing" >
+          <item row="4" column="0">
+           <layout class="QVBoxLayout">
+            <property name="spacing">
              <number>6</number>
             </property>
-            <property name="margin" >
+            <property name="margin">
              <number>0</number>
             </property>
             <item>
-             <widget class="QCheckBox" name="pronounceWordBox" >
-              <property name="text" >
+             <widget class="QCheckBox" name="pronounceWordBox">
+              <property name="text">
                <string>Pronounce the word</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
-          <item row="3" column="0" >
-           <layout class="QHBoxLayout" >
-            <property name="spacing" >
+          <item row="3" column="0">
+           <layout class="QHBoxLayout">
+            <property name="spacing">
              <number>6</number>
             </property>
-            <property name="margin" >
+            <property name="margin">
              <number>0</number>
             </property>
             <item>
-             <widget class="QLabel" name="label_4" >
-              <property name="text" >
+             <widget class="QLabel" name="label_4">
+              <property name="text">
                <string>Timeout before hide after mouse over</string>
               </property>
              </widget>
             </item>
             <item>
-             <widget class="QDoubleSpinBox" name="timeoutBeforeHideSpin" >
-              <property name="specialValueText" >
+             <widget class="QDoubleSpinBox" name="timeoutBeforeHideSpin">
+              <property name="specialValueText">
                <string>Don't hide</string>
               </property>
-              <property name="suffix" >
+              <property name="suffix">
                <string> sec</string>
               </property>
-              <property name="decimals" >
+              <property name="decimals">
                <number>1</number>
               </property>
-              <property name="minimum" >
+              <property name="minimum">
                <double>0.000000000000000</double>
               </property>
-              <property name="maximum" >
+              <property name="maximum">
                <double>10.000000000000000</double>
               </property>
-              <property name="singleStep" >
+              <property name="singleStep">
                <double>0.100000000000000</double>
               </property>
-              <property name="value" >
+              <property name="value">
                <double>0.500000000000000</double>
               </property>
              </widget>
             </item>
            </layout>
           </item>
-          <item row="2" column="0" >
-           <layout class="QHBoxLayout" >
-            <property name="spacing" >
+          <item row="2" column="0">
+           <layout class="QHBoxLayout">
+            <property name="spacing">
              <number>6</number>
             </property>
-            <property name="margin" >
+            <property name="margin">
              <number>0</number>
             </property>
             <item>
-             <widget class="QCheckBox" name="useScanModifierBox" >
-              <property name="enabled" >
+             <widget class="QCheckBox" name="useScanModifierBox">
+              <property name="enabled">
                <bool>true</bool>
               </property>
-              <property name="text" >
+              <property name="text">
                <string>Show only if modifier pressed</string>
               </property>
              </widget>
             </item>
             <item>
-             <widget class="QComboBox" name="modifierKeyBox" >
-              <property name="enabled" >
+             <widget class="QComboBox" name="modifierKeyBox">
+              <property name="enabled">
                <bool>false</bool>
               </property>
               <item>
-               <property name="text" >
+               <property name="text">
                 <string>Alt</string>
                </property>
               </item>
               <item>
-               <property name="text" >
+               <property name="text">
                 <string>Control</string>
                </property>
               </item>
               <item>
-               <property name="text" >
+               <property name="text">
                 <string>Shift</string>
                </property>
               </item>
               <item>
-               <property name="text" >
+               <property name="text">
                 <string>Win</string>
                </property>
               </item>
@@ -414,19 +388,19 @@
             </item>
            </layout>
           </item>
-          <item row="0" column="0" >
-           <widget class="QCheckBox" name="useScanBox" >
-            <property name="text" >
+          <item row="0" column="0">
+           <widget class="QCheckBox" name="useScanBox">
+            <property name="text">
              <string>Scan selection</string>
             </property>
-            <property name="checked" >
+            <property name="checked">
              <bool>true</bool>
             </property>
            </widget>
           </item>
-          <item row="1" column="0" >
-           <widget class="QCheckBox" name="showIfNotFoundBox" >
-            <property name="text" >
+          <item row="1" column="0">
+           <widget class="QCheckBox" name="showIfNotFoundBox">
+            <property name="text">
              <string>Show if word not found</string>
             </property>
            </widget>
@@ -434,108 +408,128 @@
          </layout>
         </widget>
        </item>
-       <item row="1" column="0" >
-        <widget class="QGroupBox" name="groupBox_2" >
-         <property name="title" >
-          <string>Apperance</string>
+       <item row="1" column="0">
+        <spacer>
+         <property name="orientation">
+          <enum>Qt::Vertical</enum>
+         </property>
+         <property name="sizeHint" stdset="0">
+          <size>
+           <width>20</width>
+           <height>101</height>
+          </size>
          </property>
-         <layout class="QGridLayout" >
-          <property name="margin" >
+        </spacer>
+       </item>
+      </layout>
+     </widget>
+     <widget class="QWidget" name="tabAppearance">
+      <attribute name="title">
+       <string>View</string>
+      </attribute>
+      <layout class="QVBoxLayout" name="verticalLayout_3">
+       <item>
+        <widget class="QGroupBox" name="groupBox_4">
+         <property name="title">
+          <string>View</string>
+         </property>
+         <layout class="QGridLayout" name="_4">
+          <property name="margin">
            <number>9</number>
           </property>
-          <property name="spacing" >
+          <property name="spacing">
            <number>6</number>
           </property>
-          <item row="2" column="0" >
-           <layout class="QHBoxLayout" >
-            <property name="spacing" >
+          <item row="2" column="0">
+           <layout class="QHBoxLayout" name="_5">
+            <property name="spacing">
              <number>6</number>
             </property>
-            <property name="margin" >
+            <property name="margin">
              <number>0</number>
             </property>
             <item>
-             <widget class="QLabel" name="label_3" >
-              <property name="text" >
+             <widget class="QLabel" name="label_8">
+              <property name="text">
                <string>Opacity</string>
               </property>
              </widget>
             </item>
             <item>
-             <widget class="QSpinBox" name="popupOpacitySpin" >
-              <property name="suffix" >
+             <widget class="QSpinBox" name="popupOpacitySpin">
+              <property name="suffix">
                <string>%</string>
               </property>
-              <property name="prefix" >
+              <property name="prefix">
                <string/>
               </property>
-              <property name="minimum" >
+              <property name="minimum">
                <number>1</number>
               </property>
-              <property name="maximum" >
+              <property name="maximum">
                <number>100</number>
               </property>
-              <property name="value" >
+              <property name="value">
                <number>100</number>
               </property>
              </widget>
             </item>
            </layout>
           </item>
-          <item row="0" column="0" >
-           <layout class="QHBoxLayout" >
-            <property name="spacing" >
+          <item row="0" column="0">
+           <layout class="QHBoxLayout" name="_6">
+            <property name="spacing">
              <number>6</number>
             </property>
-            <property name="margin" >
+            <property name="margin">
              <number>0</number>
             </property>
             <item>
-             <widget class="QLabel" name="label_5" >
-              <property name="text" >
+             <widget class="QLabel" name="label_9">
+              <property name="text">
                <string>Default width</string>
               </property>
              </widget>
             </item>
             <item>
-             <widget class="QSpinBox" name="popupDefaultWidthSpin" >
-              <property name="minimum" >
+             <widget class="QSpinBox" name="popupDefaultWidthSpin">
+              <property name="minimum">
                <number>96</number>
               </property>
-              <property name="maximum" >
+              <property name="maximum">
                <number>9999</number>
               </property>
-              <property name="value" >
+              <property name="value">
                <number>320</number>
               </property>
              </widget>
             </item>
            </layout>
           </item>
-          <item row="1" column="0" >
-           <layout class="QHBoxLayout" >
-            <property name="spacing" >
+          <item row="1" column="0">
+           <layout class="QHBoxLayout" name="_7">
+            <property name="spacing">
              <number>6</number>
             </property>
-            <property name="margin" >
+            <property name="margin">
              <number>0</number>
             </property>
             <item>
-             <widget class="QLabel" name="label_6" >
-              <property name="text" >
+             <widget class="QLabel" name="label_10">
+              <property name="text">
                <string>Default height</string>
               </property>
              </widget>
             </item>
             <item>
-             <widget class="QSpinBox" name="popupDefaultHeightSpin" >
-              <property name="minimum" >
+             <widget class="QSpinBox" name="popupDefaultHeightSpin">
+              <property name="minimum">
                <number>96</number>
               </property>
-              <property name="maximum" >
+              <property name="maximum">
                <number>9999</number>
               </property>
-              <property name="value" >
+              <property name="value">
                <number>240</number>
               </property>
              </widget>
@@ -545,41 +539,40 @@
          </layout>
         </widget>
        </item>
-       <item row="2" column="0" >
-        <spacer>
-         <property name="orientation" >
-          <enum>Qt::Vertical</enum>
-         </property>
-         <property name="sizeHint" stdset="0" >
-          <size>
-           <width>20</width>
-           <height>101</height>
-          </size>
-         </property>
-        </spacer>
-       </item>
       </layout>
      </widget>
-     <widget class="QWidget" name="tab" >
-      <property name="geometry" >
-       <rect>
-        <x>0</x>
-        <y>0</y>
-        <width>640</width>
-        <height>409</height>
-       </rect>
-      </property>
-      <attribute name="title" >
+     <widget class="QWidget" name="tab">
+      <attribute name="title">
        <string>Apperance</string>
       </attribute>
-      <layout class="QGridLayout" >
-       <item row="0" column="0" >
-        <widget class="QStarDict::CSSEdit" native="1" name="apperanceCSSEdit" />
+      <layout class="QGridLayout">
+       <item row="0" column="0">
+        <widget class="QStarDict::CSSEdit" name="apperanceCSSEdit" native="true">
+         <property name="sizePolicy">
+          <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
+           <horstretch>0</horstretch>
+           <verstretch>0</verstretch>
+          </sizepolicy>
+         </property>
+        </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
+   <item row="1" column="0">
+    <widget class="QDialogButtonBox" name="buttonBox">
+     <property name="orientation">
+      <enum>Qt::Horizontal</enum>
+     </property>
+     <property name="standardButtons">
+      <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
+     </property>
+     <property name="centerButtons">
+      <bool>false</bool>
+     </property>
+    </widget>
+   </item>
   </layout>
  </widget>
  <customwidgets>
@@ -599,21 +592,16 @@
   <tabstop>dictsMoveUpButton</tabstop>
   <tabstop>dictsMoveDownButton</tabstop>
   <tabstop>dictsShowInfoButton</tabstop>
-  <tabstop>pluginsTableView</tabstop>
-  <tabstop>pluginsShowInfoButton</tabstop>
-  <tabstop>pluginsConfigureButton</tabstop>
   <tabstop>useScanBox</tabstop>
   <tabstop>showIfNotFoundBox</tabstop>
   <tabstop>useScanModifierBox</tabstop>
   <tabstop>modifierKeyBox</tabstop>
   <tabstop>timeoutBeforeHideSpin</tabstop>
   <tabstop>pronounceWordBox</tabstop>
-  <tabstop>popupDefaultWidthSpin</tabstop>
-  <tabstop>popupDefaultHeightSpin</tabstop>
-  <tabstop>popupOpacitySpin</tabstop>
  </tabstops>
  <resources>
-  <include location="pixmaps/pixmaps.qrc" />
+  <include location="../plugins/stardict/pixmaps/pixmaps.qrc"/>
+  <include location="pixmaps/pixmaps.qrc"/>
  </resources>
  <connections>
   <connection>
@@ -622,11 +610,11 @@
    <receiver>QStarDict::SettingsDialog</receiver>
    <slot>accept()</slot>
    <hints>
-    <hint type="sourcelabel" >
-     <x>330</x>
-     <y>587</y>
+    <hint type="sourcelabel">
+     <x>310</x>
+     <y>230</y>
     </hint>
-    <hint type="destinationlabel" >
+    <hint type="destinationlabel">
      <x>410</x>
      <y>299</y>
     </hint>
@@ -638,11 +626,11 @@
    <receiver>QStarDict::SettingsDialog</receiver>
    <slot>reject()</slot>
    <hints>
-    <hint type="sourcelabel" >
-     <x>406</x>
-     <y>587</y>
+    <hint type="sourcelabel">
+     <x>310</x>
+     <y>230</y>
     </hint>
-    <hint type="destinationlabel" >
+    <hint type="destinationlabel">
      <x>410</x>
      <y>299</y>
     </hint>
@@ -654,13 +642,13 @@
    <receiver>modifierKeyBox</receiver>
    <slot>setEnabled(bool)</slot>
    <hints>
-    <hint type="sourcelabel" >
-     <x>165</x>
-     <y>140</y>
+    <hint type="sourcelabel">
+     <x>194</x>
+     <y>85</y>
     </hint>
-    <hint type="destinationlabel" >
-     <x>273</x>
-     <y>137</y>
+    <hint type="destinationlabel">
+     <x>289</x>
+     <y>85</y>
     </hint>
    </hints>
   </connection>
diff --git a/qstardict/speaker.cpp b/qstardict/speaker.cpp
index ef31f03..723d4c8 100644
--- a/qstardict/speaker.cpp
+++ b/qstardict/speaker.cpp
@@ -29,7 +29,7 @@ Speaker::Speaker()
 {
     m_speechProcess = new QProcess;
     QSettings settings;
-    m_speechCmd = settings.value("Speaker/speechCmd", "festival --tts").toString();
+    m_speechCmd = settings.value("Speaker/speechCmd", "flite").toString();
 }
 
 Speaker::~Speaker()
