This is from alex4src_data.zip directory 'extras'
Index: alex4src/src/main.c
===================================================================
--- alex4src.orig/src/main.c	2003-07-26 12:53:52.000000000 +0200
+++ alex4src/src/main.c	2011-06-06 22:51:43.000000000 +0200
@@ -840,6 +840,7 @@
 			i = DIGI_AUTODETECT;
 			log2file("  DIGI_AUTODETECT selected (%d)", i);
 			break;
+#ifdef ALLEGRO_WINDOWS
 		case 2:
 			i = DIGI_DIRECTX(0);
 			log2file("  DIGI_DIRECTX(0) selected (%d)", i);
@@ -848,6 +849,20 @@
 			i = DIGI_DIRECTAMX(0);
 			log2file("  DIGI_DIRECTAMX(0) selected (%d)", i);
 			break;
+#elif defined ALLEGRO_UNIX
+#ifdef DIGI_OSS
+		case 2:
+			i = DIGI_OSS;
+			log2file("  DIGI_OSS selected (%d)", i);
+			break;
+#endif
+#ifdef DIGI_ALSA
+		case 3:
+			i = DIGI_ALSA;
+			log2file("  DIGI_ALSA selected (%d)", i);
+			break;
+#endif
+#endif
 		default:
 			i = -770405;	// dummy number
 			got_sound = 0;
Index: alex4src/src/main.h
===================================================================
--- alex4src.orig/src/main.h	2003-07-26 12:53:52.000000000 +0200
+++ alex4src/src/main.h	2011-06-06 22:51:43.000000000 +0200
@@ -90,8 +90,7 @@
 
 
 // a little bounding box quickie
-#define check_bb_collision(x1,y1,w1,h1,x2,y2,w2,h2) (!( ((x1)>=(x2)+(w2)) || ((x2)>=(x1)+(w1)) || \
-                                                        ((y1)>=(y2)+(h2)) || ((y2)>=(y1)+(h1)) ))
+#define check_bb_collision(x1,y1,w1,h1,x2,y2,w2,h2) (!( ((x1)>=(x2)+(w2)) || ((x2)>=(x1)+(w1)) || ((y1)>=(y2)+(h2)) || ((y2)>=(y1)+(h1)) ))
 
 
 #endif
\ No newline at end of file
Index: alex4src/src/shooter.c
===================================================================
--- alex4src.orig/src/shooter.c	2003-07-26 12:53:52.000000000 +0200
+++ alex4src/src/shooter.c	2011-06-06 22:51:43.000000000 +0200
@@ -239,8 +239,7 @@
 
 // PIXEL PERFECT COLLISION ROUTINES COURTESY OF IVAN BALDO'S PPCOL
 
-#define pp_check_bb_collision_general(x1,y1,w1,h1,x2,y2,w2,h2) (!( ((x1)>=(x2)+(w2)) || ((x2)>=(x1)+(w1)) || \
-															((y1)>=(y2)+(h2)) || ((y2)>=(y1)+(h1)) ))
+#define pp_check_bb_collision_general(x1,y1,w1,h1,x2,y2,w2,h2) (!( ((x1)>=(x2)+(w2)) || ((x2)>=(x1)+(w1)) || ((y1)>=(y2)+(h2)) || ((y2)>=(y1)+(h1)) ))
 #define pp_check_bb_collision(mask1,mask2,x1,y1,x2,y2) pp_check_bb_collision_general(x1,y1,mask1->w,mask1->h,x2,y2,mask2->w,mask2->h)
 
 int s_check_pp_collision(BITMAP *spr1, BITMAP *spr2, int x1, int y1, int x2, int y2) {
