diff -ur qball-1.1.or/QBall/main.cpp qball-1.1/QBall/main.cpp
--- qball-1.1.or/QBall/main.cpp	2010-12-14 21:13:48.387821716 +0300
+++ qball-1.1/QBall/main.cpp	2010-12-14 21:16:23.749898252 +0300
@@ -1,6 +1,10 @@
 #include <QtGui/QApplication>
 #include "mainwindow.h"
 
+#if defined(Q_WS_QWS)
+#include <QtGui/QWSServer>
+#endif
+
 int main(int argc, char *argv[])
 {
     QApplication a(argc, argv);
@@ -14,6 +18,9 @@
     //w.show();
     w.showFullScreen();
 #endif
+#if defined(Q_WS_QWS)
+    QWSServer::setCursorVisible(false);
+#endif
     //w.init();
     //Gamestate *gamestate = new PlayingGamestate(&w);
     Gamestate *gamestate = new OptionsGamestate(&w);
