diff -ur libnsfb.orig/src/surface/Makefile libnsfb/src/surface/Makefile
--- libnsfb.orig/src/surface/Makefile	2010-11-20 13:13:46.000000000 +0300
+++ libnsfb/src/surface/Makefile	2010-11-20 13:42:01.000000000 +0300
@@ -1,4 +1,4 @@
 # Sources
-DIR_SOURCES := surface.c sdl.c vnc.c able.c ram.c linux.c x.c
+DIR_SOURCES := surface.c sdl.c ram.c linux.c
 
 include build/makefiles/Makefile.subdir
Только в libnsfb/src/surface: Makefile~
