Index: Alpy-0.1.5/_alpymodule.c
===================================================================
--- Alpy-0.1.5.orig/_alpymodule.c	2012-03-30 02:24:26.096706083 +0200
+++ Alpy-0.1.5/_alpymodule.c	2012-04-01 23:30:51.281818550 +0200
@@ -988,7 +988,7 @@
     char* name;
     int i;
 
-    if(!PyArg_ParseTuple(args, "et", &name)) {
+    if(!PyArg_ParseTuple(args, "et", _utf8, &name)) {
         return NULL;
     }
 
@@ -1194,7 +1194,7 @@
             break;
         }
     }
-    if(i == -1) {
+    if(index == -1) {
         PyErr_Format(PyExc_NameError, "no such counter: %s", name);
         PyMem_Free(name);
         return NULL;
