diff --git a/sources/Makefile b/sources/Makefile
index d0f9569..a059412 100644
--- a/sources/Makefile
+++ b/sources/Makefile
@@ -15,15 +15,15 @@
 #
 
 # The following five variables may need to be changed
-LIBRARIES=/usr/lib
-INCLUDES=/usr/include
-BINARIES=/usr/local/bin
-CC=$(CROSS)gcc
-CFLAGS= -DUNIX -DZAURUS -fsigned-char -Wall -O3 -s
+LIBRARIES?=/usr/lib
+INCLUDES?=/usr/include
+BINARIES?=/usr/local/bin
+CC?=$(CROSS)gcc
+CFLAGS:=$(CFLAGS) -DUNIX -DZAURUS -fsigned-char -Wall -O3 -s
 
 # Cross compilation variables
-LD=$(CROSS)ld
-AR=$(CROSS)ar
+LD?=$(CROSS)ld
+AR?=$(CROSS)ar
 
 OBJ4TH = errs_4th.o save_4th.o load_4th.o free_4th.o comp_4th.o \
 	name_4th.o dump_4th.o exec_4th.o open_4th.o cgen_4th.o
