diff --git a/src/fileimpl.cpp b/src/fileimpl.cpp
index 08edcb9..db8cb7f 100644
--- a/src/fileimpl.cpp
+++ b/src/fileimpl.cpp
@@ -35,6 +35,7 @@
 #include <unistd.h>
 #include <errno.h>
 #include <string.h>
+#include <cstdio>
 
 namespace cxxtools {
 
diff --git a/src/http/messageheader.cpp b/src/http/messageheader.cpp
index 9381be7..21ee7c3 100644
--- a/src/http/messageheader.cpp
+++ b/src/http/messageheader.cpp
@@ -30,6 +30,7 @@
 #include <cxxtools/clock.h>
 #include <cctype>
 #include <sstream>
+#include <cstdio>
 
 namespace cxxtools {
 
