diff -urN sarien-0.7.0/etc/sarien.cfg sarien-0.7.0-nanonote/etc/sarien.cfg
--- sarien-0.7.0/etc/sarien.cfg	2001-11-06 10:49:58.000000000 +0100
+++ sarien-0.7.0-nanonote/etc/sarien.cfg	2010-06-21 00:21:42.000000000 +0200
@@ -6,8 +6,7 @@
 # to the operating system being used.
 #
 # In UNIX:
-#	$HOME/.sarienrc
-#	/etc/sarien.conf
+#	/usr/share/sarien/sarien.cfg
 #
 # In Cygwin:
 #	/etc/sarien.conf
diff -urN sarien-0.7.0/Makefile sarien-0.7.0-nanonote/Makefile
--- sarien-0.7.0/Makefile	2001-09-04 04:17:00.000000000 +0200
+++ sarien-0.7.0-nanonote/Makefile	2010-06-21 01:29:46.000000000 +0200
@@ -72,6 +72,17 @@
 	rm -Rf $(DIST)
 	sync
 
+ipkg:
+	rm -Rf ipkg-tmp $(DIST).ipk
+	mkdir -p ipkg-tmp/CONTROL
+	cp openwrt/control ipkg-tmp/CONTROL
+	$(MAKE) DISTDIR=ipkg-tmp install
+	ipkg-build -o root -g root ipkg-tmp
+
+install: all
+	install -D -m 0755 -s --strip-program=$(STRIP) bin/sarien $(DISTDIR)$(BIN_DIR)/sarien
+	install -D -m 0644 etc/sarien.cfg $(DISTDIR)$(DATADIR)/sarien.cfg
+
 chkoldver:
 	@if [ "$(OLDVER)" = "" ]; then \
 	    echo "parameter missing: OLDVER=<old_version>"; \
diff -urN sarien-0.7.0/openwrt/control sarien-0.7.0-nanonote/openwrt/control
--- sarien-0.7.0/openwrt/control	1970-01-01 01:00:00.000000000 +0100
+++ sarien-0.7.0-nanonote/openwrt/control	2010-06-21 00:54:37.000000000 +0200
@@ -0,0 +1,18 @@
+Package: sarien
+Version: 0.7.0-1
+Source: http://sourceforge.net/projects/sarien/
+Architecture: xburst
+Maintainer: Magnus Bergman <magnusbe@algonet.se>
+Priority: optional
+Section: extras
+Depends: libsdl
+Description: An interpreter for AGI resources
+ The old Sierra games are made with the AGI engine. Sarien 
+ is a free interpreter for AGI resources.
+ .
+ Games written in AGI includes:
+   * Kings Quest I - IV
+   * Police Quest
+   * Leisure Suit Larry
+ .
+ You need the files from the original games.
diff -urN sarien-0.7.0/Rules.in sarien-0.7.0-nanonote/Rules.in
--- sarien-0.7.0/Rules.in	2001-09-06 14:41:36.000000000 +0200
+++ sarien-0.7.0-nanonote/Rules.in	2010-06-21 00:14:38.000000000 +0200
@@ -3,6 +3,7 @@
 prefix		= @prefix@
 exec_prefix 	= @prefix@
 BIN_DIR		= @bindir@
+DATA_DIR	= @datadir@/sarien
 LIB_DIR		= @libdir@
 MAN_DIR		= @mandir@/man1
 
diff -urN sarien-0.7.0/src/filesys/unix/path.c sarien-0.7.0-nanonote/src/filesys/unix/path.c
--- sarien-0.7.0/src/filesys/unix/path.c	2001-08-05 01:59:48.000000000 +0200
+++ sarien-0.7.0-nanonote/src/filesys/unix/path.c	2010-06-21 00:57:09.000000000 +0200
@@ -46,6 +46,6 @@
 #endif
 #endif
 
-	return "/etc/sarien.conf";
+	return DATADIR "/sarien.cfg";
 }
 
diff -urN sarien-0.7.0/src/graphics/sdl/sdl.c sarien-0.7.0-nanonote/src/graphics/sdl/sdl.c
--- sarien-0.7.0/src/graphics/sdl/sdl.c	2010-06-19 17:41:56.000000000 +0200
+++ sarien-0.7.0-nanonote/src/graphics/sdl/sdl.c	2010-06-20 23:02:18.000000000 +0200
@@ -460,6 +460,7 @@
 			SDL_GetError ());
 		return err_Unk;
 	}
+	SDL_ShowCursor (SDL_DISABLE);
 
 	mode = SDL_SWSURFACE | SDL_ANYFORMAT;
 
