diff --git a/src/lingot-gui-mainframe.c b/src/lingot-gui-mainframe.c
index 30c1b69..a6b35f5 100644
--- a/src/lingot-gui-mainframe.c
+++ b/src/lingot-gui-mainframe.c
@@ -107,9 +107,10 @@ void lingot_gui_mainframe_callback_about(GtkWidget* w, LingotMainFrame* frame) {
 
 void lingot_gui_mainframe_callback_view_spectrum(GtkWidget* w,
 		LingotMainFrame* frame) {
-	gtk_widget_set_visible(frame->spectrum_frame,
-			gtk_check_menu_item_get_active(
-					GTK_CHECK_MENU_ITEM(frame->view_spectrum_item)));
+	if (gtk_check_menu_item_get_active(
+				GTK_CHECK_MENU_ITEM(frame->view_spectrum_item))) {
+		gtk_widget_show(frame->spectrum_frame);
+	}
 }
 
 void lingot_gui_mainframe_callback_config_dialog(GtkWidget* w,
