Index: gnuplot-4.4.0/term/linux.trm
===================================================================
--- gnuplot-4.4.0.orig/term/linux.trm	2011-01-06 16:19:41.000000000 +0100
+++ gnuplot-4.4.0/term/linux.trm	2011-01-06 16:20:00.000000000 +0100
@@ -120,7 +120,9 @@
     char line[256];
     FILE *pipe;
 
-    LINUX_graphics_allowed = FALSE;
+    /* enabling this.  No need to be paranoid on openwrt.  popen() below won't
+       even work */
+    LINUX_graphics_allowed = TRUE;
 
     if (geteuid() != 0)
 	return;			/* if we aren't root, we cannot init graphics */
