Index: gnuplot-4.4.0/term/linux.trm
===================================================================
--- gnuplot-4.4.0.orig/term/linux.trm	2011-01-09 10:20:04.000000000 +0100
+++ gnuplot-4.4.0/term/linux.trm	2011-01-09 10:45:51.000000000 +0100
@@ -241,7 +241,7 @@
 	linetype = LT_BLACK;
     if (linetype >= 13)
 	linetype %= 13;
-    vga_setcolor(vgacolor[linetype + 2]);
+    vga_setegacolor(vgacolor[linetype + 2]);
 }
 
 TERM_PUBLIC void
