Index: gnuplot-4.4.0/src/term.c
===================================================================
--- gnuplot-4.4.0.orig/src/term.c	2011-01-06 17:10:50.000000000 +0100
+++ gnuplot-4.4.0/src/term.c	2011-01-06 17:13:37.000000000 +0100
@@ -1792,7 +1792,7 @@
    LINUX_setup has failed, also if we are logged in by network */
 #ifdef LINUXVGA
 	if (LINUX_graphics_allowed)
-#ifdef VGAGL
+#if defined(VGAGL) && defined (THREEDKIT)
 	    term_name = "vgagl";
 #else
 	    term_name = "linux";
