diff --git a/Makefile b/Makefile
index ff2e231..8414bae 100644
--- a/Makefile
+++ b/Makefile
@@ -1,7 +1,7 @@
 VERSION=$(shell if test -f manifest.uuid; then cat manifest.uuid | cut -c -10 ; else echo undefined; fi)
 
 offrss: offrss.c server.c encoding.c util.c
-	gcc -DVERSION="$(VERSION)" -g -o $@ $^ -lmrss -lcurl
+	$(CC) $(CFLAGS) $(LDFLAGS) -DVERSION="$(VERSION)" -g -o $@ $^ -lmrss -lcurl
 
 offrss.c: offrss.h
 server.c: offrss.h
