Index: ase-0.8.2/src/widgets/groupbut.cpp
===================================================================
--- ase-0.8.2.orig/src/widgets/groupbut.cpp	2010-11-28 18:47:26.000000000 +0100
+++ ase-0.8.2/src/widgets/groupbut.cpp	2011-06-13 11:33:18.000000000 +0200
@@ -101,7 +101,7 @@
   JWidget sel = find_selected(group);
 
   if (sel)
-    return (int)sel->user_data[1];
+    return (long)sel->user_data[1];
   else
     return -1;
 }
@@ -110,7 +110,7 @@
 {
   JWidget sel = find_selected(group);
 
-  if (!sel || (int)sel->user_data[1] != index) {
+  if (!sel || (long)sel->user_data[1] != index) {
     sel->setSelected(false);
     select_button(group, index);
   }
@@ -137,7 +137,7 @@
   JLink link;
 
   if (widget->type == JI_RADIO) {
-    if ((int)widget->user_data[1] == index) {
+    if ((long)widget->user_data[1] == index) {
       widget->setSelected(true);
       return true;
     }
