Index: emacs23.2-openwrt/lisp/loadup.el
===================================================================
--- emacs23.2-openwrt/lisp/loadup.el	(.../emacs-openwrt/lisp)	(revision 357)
+++ emacs23.2-openwrt/lisp/loadup.el	(.../nanonote/emacs23.2-openwrt/lisp)	(working copy)
@@ -47,6 +47,14 @@
 
 ;;; Code:
 
+;; This reduces memory usage by a few MB.  So far no negative consequences
+;; have been witnessed.
+(setq inhibit-load-charset-map t)
+
+;; Purify does in fact _safe_ memory (more efficient than normal' heap?)  so
+;; do not disable it for now.
+;(setq purify-flag nil)
+
 ;; Add subdirectories to the load-path for files that might get
 ;; autoloaded when bootstrapping.
 (if (or (equal (nth 3 command-line-args) "bootstrap")
