Index: emacs/src/emacs.c
===================================================================
--- emacs.orig/src/emacs.c	2010-12-18 20:37:40.000000000 +0100
+++ emacs/src/emacs.c	2010-12-18 20:37:40.000000000 +0100
@@ -1735,6 +1735,10 @@
 #endif  /* HAVE_NTGUI */
     }
 
+  /* syms_of_callproc() resets process-environment to nil; generating it again
+   * here, as workaround (probably only hits us since we're not dumping */
+  set_initial_environment();
+
   init_charset ();
 
   init_editfns (); /* init_process uses Voperating_system_release. */
