From 01cd959ba71d577c8a13efbdc868b46515c88334 Mon Sep 17 00:00:00 2001
From: Xiangfu <xiangfu@openmobilefree.net>
Date: Sun, 19 Feb 2012 22:20:22 +0800
Subject: [PATCH 1/2] add-MIPSED-to-icarus-for-BIG_ENDIAN.patch

---
 icarus.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/icarus.c b/icarus.c
index 48dabe5..c4c3736 100644
--- a/icarus.c
+++ b/icarus.c
@@ -290,7 +290,7 @@ static uint64_t icarus_scanhash(struct thr_info *thr, struct work *work,
         if (nonce == 0 && ret)
                 return 0xffffffff;
 
-#ifndef __BIG_ENDIAN__
+#if !defined (__BIG_ENDIAN__) && !defined(MIPSEB)
 	nonce = swab32(nonce);
 #endif
 	work->blk.nonce = 0xffffffff;
-- 
1.7.5.4

