From bed5510daca0a751e264da14367d3389f4f93dbd Mon Sep 17 00:00:00 2001
From: Xiangfu <xiangfu@openmobilefree.net>
Date: Sun, 19 Feb 2012 22:20:22 +0800
Subject: [PATCH 2/2] add-MIPSED-to-icarus-for-BIG_ENDIAN.patch

---
 icarus.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/icarus.c b/icarus.c
index 1d59657..26f4927 100644
--- a/icarus.c
+++ b/icarus.c
@@ -287,7 +287,7 @@ static uint64_t icarus_scanhash(struct thr_info *thr, struct work *work,
         if (nonce == 0 && ret)
                 return 0xffffffff;
 
-#ifndef __BIG_ENDIAN__
+#if !defined (__BIG_ENDIAN__) && !defined(MIPSEB)
 	nonce = swab32(nonce);
 #endif
 	work->blk.nonce = 0xffffffff;
-- 
1.7.5.4

