From b1785d1f07d9482b5ed6104c87b39a3888fa8efb Mon Sep 17 00:00:00 2001
From: Xiangfu <xiangfu@openmobilefree.net>
Date: Fri, 11 May 2012 11:04:28 +0800
Subject: [PATCH 2/2] add-MIPSEB-to-icarus-for-BIG_ENDIAN

---
 driver-icarus.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/driver-icarus.c b/driver-icarus.c
index dba6617..5ef3300 100644
--- a/driver-icarus.c
+++ b/driver-icarus.c
@@ -350,7 +350,7 @@ static uint64_t icarus_scanhash(struct thr_info *thr, struct work *work,
 		return ESTIMATE_HASHES;
 	}
 
-#ifndef __BIG_ENDIAN__
+#if !defined (__BIG_ENDIAN__) && !defined(MIPSEB)
 	nonce = swab32(nonce);
 #endif
 
-- 
1.7.5.4

