From aed89bc1d85f6d4dc0834f850730894571ed9e71 Mon Sep 17 00:00:00 2001
From: Xiangfu <xiangfu@openmobilefree.net>
Date: Tue, 14 Feb 2012 00:02:06 +0800
Subject: [PATCH 4/6] update the message fit both bitforce and Icarus


Signed-off-by: Xiangfu <xiangfu@openmobilefree.net>
---
 cgminer.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/cgminer.c b/cgminer.c
index cea6f59..bfb8399 100644
--- a/cgminer.c
+++ b/cgminer.c
@@ -780,7 +780,7 @@ static struct opt_table opt_config_table[] = {
 #if defined(USE_BITFORCE) || (defined(USE_ICARUS) && defined(WIN32))
 	OPT_WITH_ARG("--scan-serial|-S",
 		     add_serial, NULL, NULL,
-		     "Serial port to probe for BitForce device"),
+		     "Serial port to probe for FPGA Mining device"),
 #endif
 	OPT_WITH_ARG("--scan-time|-s",
 		     set_int_0_to_9999, opt_show_intval, &opt_scantime,
-- 
1.7.5.4

