From beaf158a409920ab436cef749b997f487e0225b9 Mon Sep 17 00:00:00 2001
From: Xiangfu <xiangfu@openmobilefree.net>
Date: Tue, 14 Feb 2012 00:01:20 +0800
Subject: [PATCH 5/7] update README for Icarus

---
 README |    5 +++--
 1 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/README b/README
index f3c0a26..30cf037 100644
--- a/README
+++ b/README
@@ -50,6 +50,7 @@ CGMiner specific configuration options:
 	--disable-opencl        Override detection and disable building with opencl
 	--disable-adl           Override detection and disable building with adl
 	--enable-bitforce       Compile support for BitForce FPGAs(default disabled)
+	--enable-icarus         Compile support for Icarus Board(default disabled)
 
 Basic *nix build instructions:
 	To build with GPU mining support:
@@ -193,9 +194,9 @@ GPU only options:
 --ndevs|-n          Enumerate number of detected GPUs and exit
 
 
-BitForce only options:
+FPGA mining boards(BitForce, Icarus) only options:
 
---scan-serial|-S <arg> Serial port to probe for BitForce device
+--scan-serial|-S <arg> Serial port to probe for FPGA mining device
 
 
 CPU only options:
-- 
1.7.5.4

