From 5ae847013d58990a1686d2875b768945c65a2f6e Mon Sep 17 00:00:00 2001
From: Xiangfu <xiangfu@openmobilefree.net>
Date: Tue, 14 Feb 2012 00:01:20 +0800
Subject: [PATCH 5/6] update README for Icarus


Signed-off-by: Xiangfu <xiangfu@openmobilefree.net>
---
 README |    6 ++++++
 1 files changed, 6 insertions(+), 0 deletions(-)

diff --git a/README b/README
index f3c0a26..0ef1b9f 100644
--- a/README
+++ b/README
@@ -50,6 +50,7 @@ CGMiner specific configuration options:
 	--disable-opencl        Override detection and disable building with opencl
 	--disable-adl           Override detection and disable building with adl
 	--enable-bitforce       Compile support for BitForce FPGAs(default disabled)
+	--enable-icarus         Compile support for Icarus Board(default disabled)
 
 Basic *nix build instructions:
 	To build with GPU mining support:
@@ -198,6 +199,11 @@ BitForce only options:
 --scan-serial|-S <arg> Serial port to probe for BitForce device
 
 
+Icarus only options(only under Windows system):
+
+--scan-serial|-S <arg> Serial port to probe for Icarus device
+
+
 CPU only options:
 
 --algo|-a <arg>     Specify sha256 implementation for CPU mining:
-- 
1.7.5.4

