--- sdltest-2.2.0/CMakeLists.txt	2010-08-04 11:50:25.000000000 +0200
+++ sdltest-2.2.0m/CMakeLists.txt	2010-08-04 12:23:43.000000000 +0200
@@ -1,13 +1,13 @@
 cmake_minimum_required (VERSION 2.8)
 
-set(PHYSFS_INCLUDE /usr/include/)
-set(PHYSFS_LIBRARY physfs)
-set(LUA_INCLUDE /usr/include/lua5.1)
-set(LUA_LIBRARY lua5.1)
-set(SDL_INCLUDE /usr/include/SDL)
-set(SDL_LIBRARY SDL SDL_image SDL_gfx SDL_mixer)
-set(FREETYPE_INCLUDE /usr/include/freetype2)
-set(FREETYPE_LIBRARY freetype)
+set(PHYSFS_INCLUDE $ENV{STAGING_DIR}/usr/include/)
+set(PHYSFS_LIBRARY $ENV{STAGING_DIR}/usr/lib/libphysfs.so)
+set(LUA_INCLUDE $ENV{STAGING_DIR}/usr/include/lua)
+set(LUA_LIBRARY $ENV{STAGING_DIR}/usr/lib/liblua.so crypt)
+set(SDL_INCLUDE $ENV{STAGING_DIR}/usr/include/SDL)
+set(SDL_LIBRARY $ENV{STAGING_DIR}/usr/lib/libSDL.so $ENV{STAGING_DIR}/usr/lib/libSDL_image.so $ENV{STAGING_DIR}/usr/lib/libSDL_gfx.so $ENV{STAGING_DIR}/usr/lib/libSDL_mixer.so)
+set(FREETYPE_INCLUDE $ENV{STAGING_DIR}/usr/include/freetype2)
+set(FREETYPE_LIBRARY $ENV{STAGING_DIR}/usr/lib/libfreetype.so)
 
 include_directories(${PHYSFS_INCLUDE} ${LUA_INCLUDE} ${SDL_INCLUDE} $(FREETYPE_INCLUDE})
 include_directories(".")
