From b296cf18278529fdcf820e79fe52ab1a27a19917 Mon Sep 17 00:00:00 2001
From: Xiangfu <xiangfu@openmobilefree.net>
Date: Fri, 9 Mar 2012 16:31:55 +0800
Subject: [PATCH 2/2] disable-mouse

---
 src/display.c |    4 ++++
 1 files changed, 4 insertions(+), 0 deletions(-)

diff --git a/src/display.c b/src/display.c
index c2e4d56..dddd9f0 100644
--- a/src/display.c
+++ b/src/display.c
@@ -114,6 +114,10 @@ bard_display *bard_display_open(cst_features *config)
     /* unless you move it */
     SDL_ShowCursor(SDL_DISABLE);
 
+#ifdef BARD_TARGET_BEN_NANONOTE
+    SDL_ShowCursor(SDL_DISABLE);
+#endif
+
     return display;
 }
 
-- 
1.7.5.4

