Index: allegro-4.4.2/demos/shooter/demo.c
===================================================================
--- allegro-4.4.2.orig/demos/shooter/demo.c	2011-05-29 12:56:58.000000000 +0200
+++ allegro-4.4.2/demos/shooter/demo.c	2011-05-31 15:57:29.000000000 +0200
@@ -1,3 +1,5 @@
+#define DEBUGMODE
+
 #include "demo.h"
 #include "data.h"
 #include "expl.h"
@@ -90,9 +92,9 @@
       allegro_message("Error initialising joystick\n%s\n", allegro_error);
       install_joystick(JOY_TYPE_NONE);
    }
-
-   if (set_gfx_mode(GFX_AUTODETECT, 320, 200, 0, 0) != 0) {
-      if (set_gfx_mode(GFX_SAFE, 320, 200, 0, 0) != 0) {
+   /* set_color_depth(32); */
+   if (set_gfx_mode(GFX_AUTODETECT, 320, 240, 0, 0) != 0) {
+      if (set_gfx_mode(GFX_SAFE, 320, 240, 0, 0) != 0) {
          set_gfx_mode(GFX_TEXT, 0, 0, 0, 0);
          allegro_message("Unable to set any graphic mode\n%s\n",
                          allegro_error);
