diff -ur fbida-2.07_orig/fbi.c fbida-2.07/fbi.c
--- fbida-2.07_orig/fbi.c	2008-06-09 18:53:33.000000000 +0400
+++ fbida-2.07/fbi.c	2010-10-18 22:22:33.166338378 +0400
@@ -548,7 +548,7 @@
 	L"  L           - rotate counter-clockwise",
     };
 
-    shadow_draw_text_box(face, 24, 16, transparency,
+    shadow_draw_text_box(face, 0, 0, transparency,
 			 help, ARRAY_SIZE(help));
     shadow_render();
 }
@@ -1465,7 +1465,7 @@
 
     font_init();
     if (NULL == fontname)
-	fontname = "monospace:size=16";
+	fontname = "monospace:size=10";
     face = font_open(fontname);
     if (NULL == face) {
 	fprintf(stderr,"can't open font: %s\n",fontname);
