diff -ruN a/src/conf.cpp b/src-b/conf.cpp
--- a/src/conf.cpp	2010-03-05 18:16:47.000000000 +0100
+++ b/src/conf.cpp	2010-03-05 18:16:40.000000000 +0100
@@ -56,7 +56,7 @@
 #else
 	add_entry("/apps/stardict/preferences/dictionary/use_custom_font", false);
 #endif
-	add_entry("/apps/stardict/preferences/network/enable_netdict", true);
+	add_entry("/apps/stardict/preferences/network/enable_netdict", false);
 	// Default stardictd server.
 	add_entry("/apps/stardict/preferences/network/server", std::string(_("dict.stardict.org")));
 	add_entry("/apps/stardict/preferences/network/port", 2628);
