diff --git a/avrdude.conf.in b/avrdude.conf.in
index 3e4066c..029fbb0 100644
--- a/avrdude.conf.in
+++ b/avrdude.conf.in
@@ -592,6 +592,29 @@ programmer
   type  = avr910;
 ;
 
+#
+# 8:10 card pin assignment:
+#
+# DAT1  1
+# DAT0  2
+# VSS   -
+# CLK   - (reserved for clock output)
+# VDD   -
+# CMD   5
+# DAT3  7
+# DAT2  8
+#
+
+programmer
+  id    = "nanonote_uart";
+  desc  = "NanoNote UART 8:10 card";
+  type  = nanonote;
+  reset = 8;
+  sck   = 2;
+  mosi  = 7;
+  miso  = 1;
+;
+
 @HAVE_PARPORT_BEGIN@  Inclusion of the following depends on --enable-parport
 # Parallel port programmers.
 
-- 
1.7.0.4

