diff --git a/avrdude.conf.in b/avrdude.conf.in
index 029fbb0..bcc26cd 100644
--- a/avrdude.conf.in
+++ b/avrdude.conf.in
@@ -615,6 +615,17 @@ programmer
   miso  = 1;
 ;
 
+programmer
+  id    = "nanonote_atusb";
+  desc  = "NanoNote 8:10 card adapter for ATUSB";
+  type  = nanonote;
+  reset = 1;
+  sck   = 8;
+  mosi  = 7;
+  miso  = 6;
+  pgmled = 2;
+;
+
 @HAVE_PARPORT_BEGIN@  Inclusion of the following depends on --enable-parport
 # Parallel port programmers.
 
-- 
1.7.0.4

