--- gmenu2x/src/filedialog.cpp	2010-08-03 12:18:26.000000000 +0200
+++ gmenu2xm/src/filedialog.cpp	2010-08-03 15:08:13.000000000 +0200
@@ -35,7 +35,7 @@
 	this->text = text;
 	this->filter = filter;
 	this->file = "";
-	setPath("/card");
+	setPath("/");
 	title = "File Browser";
 	if (!file.empty()) {
 		string::size_type pos = file.rfind("/");
@@ -50,7 +50,7 @@
 bool FileDialog::exec() {
 	bool close = false, result = true, ts_pressed = false;
 	if (!fileExists(path()))
-		setPath("/card");
+		setPath("/");
 
 	fl.setFilter(filter);
 	fl.browse();
@@ -155,7 +155,7 @@
 			} break;
 			case FD_ACTION_GOUP: {
 				string::size_type p = path().rfind("/");
-				if (p==string::npos || path().substr(0,11)!="/card" || p<4)
+				if (p==string::npos)
 					return false;
 				else
 					setPath( path().substr(0,p) );
--- gmenu2x/src/gmenu2x.cpp	2010-08-03 13:55:31.000000000 +0200
+++ gmenu2xm/src/gmenu2x.cpp	2010-08-03 13:55:50.000000000 +0200
@@ -998,7 +998,7 @@
 }
 
 void GMenu2X::explorer() {
-	FileDialog fd(this,tr["Select an application"],".gpu,.dge,.sh");
+	FileDialog fd(this,tr["Select an application"],".gpu,.dge,.sh,");
 	if (fd.exec()) {
 		if (confInt["saveSelection"] && (confInt["section"]!=menu->selSectionIndex() || confInt["link"]!=menu->selLinkIndex()))
 			writeConfig();
