Index: MPlayer-r33333/libmpcodecs/vd_theora.c
===================================================================
--- MPlayer-r33333.orig/libmpcodecs/vd_theora.c	2011-05-01 10:23:23.000000000 +0200
+++ MPlayer-r33333/libmpcodecs/vd_theora.c	2011-05-01 10:45:06.000000000 +0200
@@ -133,7 +133,7 @@
     mp_msg(MSGT_DECVIDEO,MSGL_V,"INFO: Theora video init ok!\n");
     mp_msg(MSGT_DECVIDEO,MSGL_INFO,"Frame: %dx%d, Picture %dx%d, Offset [%d,%d]\n", context->inf.width, context->inf.height, context->inf.frame_width, context->inf.frame_height, context->inf.offset_x, context->inf.offset_y);
 
-    return mpcodecs_config_vo (sh,context->inf.width,context->inf.height,theora_pixelformat2imgfmt(context->inf.pixelformat));
+    return mpcodecs_config_vo (sh,context->inf.frame_width,context->inf.frame_height,theora_pixelformat2imgfmt(context->inf.pixelformat));
 
 err_out:
     free(context);
