--- w3m-0.5.2/configure	2010-09-25 20:57:25.128744963 +0400
+++ w3m-0.5.2_mod/configure	2010-09-25 20:57:51.055870033 +0400
@@ -6221,7 +6221,7 @@
 _ACEOF
 
 
-for lib in bsd BSD 44bsd socket nsl dld dl
+for lib in BSD 44bsd socket nsl dld dl
 do
  lib=$lib
  { echo "$as_me:$LINENO: checking for -l$lib" >&5
@@ -7886,7 +7886,7 @@
    { (exit 1); exit 1; }; }
    fi
  fi
- unset ac_cv_lib_gc_GC_init
+
  { echo "$as_me:$LINENO: checking for GC_init in -lgc" >&5
 echo $ECHO_N "checking for GC_init in -lgc... $ECHO_C" >&6; }
 if test "${ac_cv_lib_gc_GC_init+set}" = set; then
