--- w3m-0.5.2/Makefile.in	2007-05-29 16:01:04.000000000 +0400
+++ w3m-0.5.2_mod/Makefile.in	2010-09-25 22:01:16.666250962 +0400
@@ -134,7 +134,7 @@
 $(OBJS) $(LOBJS): fm.h funcname1.h
 
 tagtable.c: tagtable.tab mktable$(EXT) html.h 
-	./mktable$(EXT) 100 $(srcdir)/tagtable.tab > $@
+#	./mktable$(EXT) 100 $(srcdir)/tagtable.tab > $@
 
 indep.o: indep.c fm.h funcname1.h
 func.o: funcname.c functable.c funcname1.h
@@ -167,7 +167,7 @@
 
 functable.c: funcname.tab mktable$(EXT)
 	sort funcname.tab | $(AWK) -f $(top_srcdir)/functable.awk > functable.tab
-	./mktable$(EXT) 100 functable.tab > $@
+#	./mktable$(EXT) 100 functable.tab > $@
 	-rm -f functable.tab
 
 mktable$(EXT): mktable.o dummy.o $(ALIB)
