diff -ur w3m-0.5.2.orig/configure w3m-0.5.2/configure
--- w3m-0.5.2.orig/configure	2011-02-25 21:51:15.648183991 +0300
+++ w3m-0.5.2/configure	2011-02-25 22:03:24.676058666 +0300
@@ -5626,7 +5626,7 @@
  else
    { echo "$as_me:$LINENO: WARNING: GdkPixbuf is not installed.  Install GdkPixbuf (version >= 2.0.0)" >&5
 echo "$as_me: WARNING: GdkPixbuf is not installed.  Install GdkPixbuf (version >= 2.0.0)" >&2;}
-   have_gdkpixbuf="no"; have_gtk2="no"
+   have_gdkpixbuf="yes"; have_gtk2="yes"
  fi
   fi
   if test x"$with_gdkpixbuf" = xyes; then
